### PR TITLE
Add names to all typedef'd structs, unions, and enums

### PR DESCRIPTION
**Pull request by cadmic**
Untested but hopefully it works

### CODE REVIEW

File: include/ultra64/gu.h [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 #define ROUND(x) (s32)(((x) >= 0.0) ? ((x) + 0.5) : ((x) - 0.5))
 
-typedef union {
+typedef union word {
```
Comment (Thar0) [inline include/ultra64/gu.h:10]:
```suggestion
typedef union du {
```

**Comment by cadmic** [comment]
This does all files in include/ and src/, but should I revert include/ultra64/*.h?

File: include/ultra64/pfs.h [hunk 2]
```diff
@@ -85,27 +85,27 @@ typedef struct {
...
+typedef struct OSPfsState {
     /* 0x00 */ u32 file_size;       /* bytes */
     /* 0x04 */ u32 game_code;
     /* 0x08 */ u16 company_code;
     /* 0x0C */ char ext_name[4];
     /* 0x10 */ char game_name[16];
 } OSPfsState; // size = 0x20
 
-typedef union {
+typedef union inode_t {
```
Comment (Thar0) [inline include/ultra64/pfs.h:96]:
```suggestion
typedef union __OSInodeUnit {
```

File: include/ultra64/thread.h [hunk 3]
```diff
@@ -27,15 +27,15 @@
 typedef s32 OSPri;
 typedef s32 OSId;
 
-typedef union {
+typedef union f {
```
Comment (Thar0) [inline include/ultra64/thread.h:30]:
```suggestion
typedef union __OSfp {
```

File: include/ultra64/xstdio.h [hunk 4]
```diff
@@ -10,7 +10,7 @@
 #define LONG_DOUBLE_TYPE long double
 #endif
 
-typedef struct {
+typedef struct v {
```
Comment (Thar0) [inline include/ultra64/xstdio.h:13]:
```suggestion
typedef struct _Pft {
```

**Review by Thar0 — COMMENTED** [review]
Some nested struct/unions didn't quite work, highlighted some specific cases

**Comment by cadmic** [comment]
Thanks, I redid it with a better regex

**Review by fig02 — APPROVED** [review]
cant say i checked every single thing, but trust that the regex did its job. if there are any issues for whatever reason we can fix them

**Review by Thar0 — APPROVED** [review]

**Comment by fig02** [comment]
merging in a few hours if no more review

**Pull request merged**